src/content/en/fundamentals/performance/webpack/use-long-term-caching.md

```diff
@@ -0,0 +1,647 @@
+
+<pre class="prettyprint">
+<strong>                           Asset   Size  Chunks             Chunk Names</strong>
+./vendor.d9e134771799ecdf9483.js  47 kB       1  [emitted]  vendor
+</pre>
+
+↓
+
+<pre class="prettyprint">
+<strong>                            Asset   Size  Chunks             Chunk Names</strong>
+./vendor.e6ea4504d61a1cc1c60b.js  47 kB       1  [emitted]  vendor
+</pre>
+
+This happens because the webpack bundle, apart from the code of modules, has [_a
+runtime_](https://webpack.js.org/concepts/manifest/) – a small piece of code that manages the
+module execution. When you split the code into multiple files, this piece of code starts including a
+mapping between chunk ids and corresponding files:
+
+    // vendor.e6ea4504d61a1cc1c60b.js
+    script.src = __webpack_require__.p + chunkId + "." + {
+      "0": "2f2269c7f0a55a5c1871"
+    }[chunkId] + ".js";
+
+Webpack includes this runtime into the last generated chunk, which is `vendor` in our case. And
+every time any chunk changes, this piece of code changes too, causing the whole <code>vendor</code>
+chunk to change.
+
+To solve this, let’s move the runtime into a separate file by creating an extra empty chunk with the
+`CommonsChunkPlugin`:
+
+    // webpack.config.js
+    module.exports = {
+      plugins: [
+        new webpack.optimize.CommonsChunkPlugin({
+          name: 'vendor',
+
+          minChunks: module => module.context &&
+            module.context.includes('node_modules'),
+        }),
+
+        // This plugin must come after the vendor one (because webpack
+        // includes runtime into the last chunk)
+        new webpack.optimize.CommonsChunkPlugin({
+          name: 'runtime',
+
+          // minChunks: Infinity means that no app modules
+          // will be included into this chunk
+          minChunks: Infinity,
+        }),
+      ],
+    };
+
+After these changes, each build will be generating three files:
+
+<pre class="prettyprint">
+$ webpack
+Hash: ac01483e8fec1fa70676
+Version: webpack 3.8.1
+Time: 3816ms
+<strong>                            Asset     Size  Chunks             Chunk Names</strong>
+   ./main.00bab6fd3100008a42b0.js    82 kB       0  [emitted]  main
+ ./vendor.26886caf15818fa82dfa.js    46 kB       1  [emitted]  vendor
+./runtime.79f17c27b335abc7aaf4.js  1.45 kB       3  [emitted]  runtime
+</pre>
+
+Include them into `index.html` in the reverse order – and you’re done:
+
+    <!-- index.html -->
+    <script src="./runtime.79f17c27b335abc7aaf4.js"></script>
+    <script src="./vendor.26886caf15818fa82dfa.js"></script>
+    <script src="./main.00bab6fd3100008a42b0.js"></script>
+
+### Further reading {: .hide-from-toc }
+
+* Webpack guide [on long term caching](https://webpack.js.org/guides/caching/)
+
+* Webpack docs [about webpack runtime and manifest](https://webpack.js.org/concepts/manifest/)
+
+* [“Getting the most out of the CommonsChunkPlugin”](https://medium.com/webpack/
```

Comment:
don't linebreak URLs, otherwise the link breaks. it's OK if this exceeds 80 chars